WalletWasabi.Fluent/ViewModels/Wallets/Send/SendViewModel.cs

```diff
@@ -81,6 +82,15 @@ public SendViewModel(Wallet wallet, IObservable<Unit> balanceChanged, Observable
 
 		PasteCommand = ReactiveCommand.CreateFromTask(async () => await OnPasteAsync());
 		AutoPasteCommand = ReactiveCommand.CreateFromTask(async () => await OnAutoPasteAsync());
+		InsertMaxCommand = ReactiveCommand.Create(() =>
+		{
+			if (Balance.BalanceBtc is not null)
+			{
+				string[] balanceStringParts = Balance.BalanceBtc.Split(" ");
+				string wholeBalance = string.Join("", balanceStringParts.SkipLast(1));
+				AmountBtc = decimal.Parse(wholeBalance);
+			}
+		});
```

Comment:
You can get the total balance simply from the wallet.

```suggestion
		InsertMaxCommand = ReactiveCommand.Create(() => AmountBtc = _wallet.Coins.TotalAmount().ToDecimal(MoneyUnit.BTC));
```